src/body.rs

```diff
@@ -117,6 +119,42 @@ fn mk_err<T>(_: T) -> Response {
     StatusCode::BAD_REQUEST.into_response()
 }
 
+pub struct MultipartForm(pub Multipart<Cursor<Vec<u8>>>);
+
+impl<S: 'static> Extract<S> for MultipartForm {
+    // Note: cannot use `existential type` here due to ICE
+    type Fut = FutureObj<'static, Result<Self, Response>>;
+
+    fn extract(data: &mut S, req: &mut Request, params: &RouteMatch<'_>) -> Self::Fut {
+        // https://stackoverflow.com/questions/43424982/how-to-parse-multipart-forms-using-abonander-multipart-with-rocket
+        let ct = match req.headers().get("content-type") {
+            Some(h) => h,
+            None => {
+                // return FutureObj::new(Box::new(Err(
+                //     http::status::StatusCode::BAD_REQUEST.into_response()
+                // )));
+                panic!("Headers have no key as content-type")
+                // TODO: How to return early here? The above return is not working, saying:
+                //   134 |                 return FutureObj::new(Box::new(Err(
+                //       |                        ^^^^^^^^^^^^^^ the trait `core::future::future::Future` is not implemented for `std::result::Result<_, http::response::Response<body::Body>>
+            }
```

Comment:
Question 1: How to return early here when the header has no `content-type` key?
I tried to do this but not working, error see in the comments in code above.
```rust
return FutureObj::new(Box::new(Err(
                http::status::StatusCode::BAD_REQUEST.into_response()
                )));
``` 🤔

Reply:
Looks like I need to use a `async { ... }`. But not sure if the current way of doing is good enough. Please give some feedback :)

Reply:
So, one thing you can do instead is the following:

```rust
const BOUNDARY = "boundary=";
let boundary = req.headers().get("content-type")
    .map(|ct| {
        let ct = ct.to_str().ok()?;
        let idx = ct.find(BOUNDARY)?;
        Some(ct[idx + "BOUNDARY".len()..].to_string())
    });
```

Then, inside the bottom `async` block, you can say:

```rust
let boundary = boundary.ok_or_else(mk_err)?;
```

Reply:
Thank you very much, this looks much better and never thought one could do this. Learned a lot :) However I need to do some minor changes on it, as the `map` call returns me an option of option, and the `ok_or_else` takes in a function that takes no arguments. 

```rust
const BOUNDARY: &str = "boundary=";
        let boundary = req.headers().get("content-type").map_or(None, |ct| {
            let ct = ct.to_str().ok()?;
            let idx = ct.find(BOUNDARY)?;
            Some(ct[idx + BOUNDARY.len()..].to_string())
        });
...

let boundary = boundary.ok_or_else(|| {
                    StatusCode::BAD_REQUEST.into_response()
                })?;
```

Pushed another commit, perhaps you can take a look again?